File: app/src/main/java/com/an/trailers/data/local/converter/GenreListTypeConverter.java
package com.an.trailers.data.local.converter;

import android.arch.persistence.room.TypeConverter;

import com.an.trailers.data.remote.model.Genre;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import java.lang.reflect.Type;
import java.util.List;

public class GenreListTypeConverter {

    @TypeConverter
    public List<Genre> fromString(String value) {
        Type listType = new TypeToken<List<Genre>>() {}.getType();
        List<Genre> genres = new Gson().fromJson(value, listType);
        return genres;
    }

    @TypeConverter
    public String fromList(List<Genre> genres) {
        return new Gson().toJson(genres);
    }
}

File: app/src/main/java/com/an/trailers/data/local/converter/GenreListTypeConverter.kt
package com.an.trailers.data.local.converter

import android.arch.persistence.room.TypeConverter
import com.an.trailers.data.remote.model.Genre
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class GenreListTypeConverter {

    @TypeConverter
    fun fromString(value: String): List<Genre>? {
        val listType = object : TypeToken<List<Genre>>() {}.type
        return Gson().fromJson<List<Genre>>(value, listType)
    }

    @TypeConverter
    fun fromList(list: List<Genre>?): String {
        val gson = Gson()
        return gson.toJson(list)
    }
}
